src/anomalib/data/datasets/base/image.py

```diff
@@ -273,23 +270,24 @@ def __getitem__(self, index: int) -> DatasetItem:
         if self.task == TaskType.CLASSIFICATION:
             item["image"] = self.augmentations(image) if self.augmentations else image
         elif self.task == TaskType.SEGMENTATION:
-            # Only Anomalous (1) images have masks in anomaly datasets
-            # Therefore, create empty mask for Normal (0) images.
+            # Create empty mask for:
+            # - Normal samples (label_index = 0)
+            # - Unknown samples (label_index = -1)
+            # Only use mask for anomalous samples (label_index = 1)
             mask = (
                 Mask(torch.zeros(image.shape[-2:])).to(torch.uint8)
-                if label_index == LabelName.NORMAL
+                if label_index in {LabelName.NORMAL, LabelName.UNKNOWN}
                 else read_mask(mask_path, as_tensor=True)
             )
             item["image"], item["gt_mask"] = self.augmentations(image, mask) if self.augmentations else (image, mask)
-
         else:
             msg = f"Unknown task type: {self.task}"
             raise ValueError(msg)
 
         return ImageItem(
             image=item["image"],
             gt_mask=item.get("gt_mask"),
-            gt_label=int(label_index),
+            gt_label=torch.tensor(label_index),  # Convert to tensor to match type hints
```

Comment:
In case of `UNKNOWN` label, the `-1` label_index value will get converted to a positive (`True`) boolean by the dataclass validator. When combined with the all-zero gt_mask that I commented on above, this leads to the weird situation where an `UNKNOWN` test sample will have a positive (anomalous) GT class label, but an empty (all-normal) GT mask. It may be better to have consistency between GT-mask and -label.